routing/segment.hpp

```diff
@@ -71,12 +79,108 @@ class Segment final
   bool m_forward = false;
 };
 
+class SegmentWeight final
+{
+public:
+  SegmentWeight() = default;
+
+  constexpr explicit SegmentWeight(double weight) : m_weight(weight) {}
```

Comment:
Do we really need explicit here? If you get rid of "explicit" keyword here, you may get rid of ```operator + (double, SegmentWeight)``` and ```operator + (SegmentWeight, double)```.